.specs/data-delete-flow/design.md

```diff
@@ -0,0 +1,882 @@
+
+**Important Constraint**: Account and Group coValues cannot be deleted. This is required to keep the tombstone permissions verifiable.
+
+The design leverages existing transaction validation mechanisms, adding delete-specific checks in `tryAddTransactions`, and implements sync restrictions to prevent deleted content from propagating while allowing delete operations to sync.
+
+## Backward compatibility (wire protocol + mixed-version peers)
+
+This design is intended to be rolled out without introducing new message types or changing the shape of existing sync messages (`load`, `known`, `content`, `done`). Instead, we change *semantics* for deleted coValues while remaining compatible with older peers.
+
+Key points:
+
+- **No protocol versioning required**: delete is represented as an ordinary unencrypted (`trusting`) transaction with meta `{ deleted: true }`, and a dedicated delete session ID (`${accountOrAgentID}_session_z${random}_deleted`, generated by `crypto.newDeleteSessionID(...)`).
+- **Mixed-version safety**:
+  - New peers must **not accept** or **re-propagate** historical content for a deleted coValue (they only accept/sync the tombstone).
+  - Old peers may continue to send historical content. New peers handle this by **ignoring** non-delete sessions for deleted coValues (see “Sync message behavior” below). This avoids resurrection even when older peers don’t understand deletion.
+- **Sync quenching (“stop syncing knownState”)**: when interacting with older peers, we prevent them from continuously trying to upload historical sessions by responding with a `known` message that merges our current knownState with the peer’s knownState (`stopSyncingKnownStateMessage(peerKnownState)`), so the sender believes we already have what it has (even if we intentionally dropped it due to deletion).
+
+## Architecture / Components
+
+### 1. Delete API Method
+
+**Location**: `packages/cojson/src/coValueCore/coValueCore.ts`
+
+**Changes**:
+- Add method `deleteCoValue()` to `CoValueCore`:
+  - Checks that the coValue is not an Account or Group (throws error if it is)
+  - Checks that the current account has admin permissions on the coValue
+  - Throws error if not admin
+  - Creates delete transaction with meta `{ deleted: true }`
+  - Uses a dedicated delete session (suffix `_deleted`, generated by `crypto.newDeleteSessionID(node.getCurrentAccountOrAgentID())`)
+
+**Key Logic (final implementation)**:
+```typescript
+/**
+ * Creates a delete marker transaction for this CoValue and sets the coValue as deleted
+ *
+ * Constraints:
+ * - Account and Group CoValues cannot be deleted.
+ * - Only admins can delete a coValue.
+ */
+deleteCoValue() {
+  if (!this.verified) {
+    throw new Error("Cannot delete coValue without verified state");
+  }
+
+  if (this.isGroupOrAccount()) {
+    throw new Error("Cannot delete Group or Account coValues");
+  }
+
+  if (this.isDeleted) {
+    throw new Error("CoValue is already deleted");
+  }
```

Comment:
What made you decide to throw an error here instead of making deletion idempotent? I've got no strong preference, but wondering what your rationale was

Reply:
Agree, I got into this error in my music player experiment and it's quite annoying.

Going to replace with an early return.